### ISSUE TITLE
Persistent Replace All tooltip

### ISSUE THREAD

**Issue by johnnyreilly** [atom/find-and-replace#501]
I've just noticed a problem which affects me when I'm attempting to find and replace in a single file (using CTRL+F key binding).

After you've hit "Replace All" to replace all instances within a file the button becomes disabled and the tooltip fails to disappear. It hangs around and obscures the buttons above it:

![persistenttooltip](https://cloud.githubusercontent.com/assets/1010525/9303182/3cfe5d9e-44d4-11e5-9bf3-3a96255b7663.jpg)

The only way to get rid of it (it seems) is to change the content of the "find" input so it matches something in the current file. Then the "Replace All" button becomes enabled once more and the tooltip stops persisting.

I'm using Windows 8.1.

**Comment by adambuczynski** [comment]
Happens on mac as well.

**Comment by iGadget73** [comment]
Why closed? Still a nagging issue.

**Comment by 50Wliu** [comment]
@iGadget73 This issue is not closed.

**Comment by iosdev-republicofapps** [comment]
Hi,

This is a critical bug that is making Atom almost unusable at times for me.  See https://github.com/atom/atom/issues/8532 .

Do we have an ETA for a fix?

Any suggestions for how to get around this in the meantime?

It's a real usability issue ...

Thanks!

**Comment by SlimeQ** [comment]
[find and replace bug](http://i.imgur.com/rCpiaTy.png)

this is so bad

**Comment by thisisnian** [comment]
Had a look at this, the issue is the tooltip isn't removed when the button is disabled.
Usually when the button is enabled it toggles depending if your mouse hovers over it.
Since it is disabled this would have zero effect hence the issue.

I had a look at removing and adding tooltips when it is disabled/enabled and I got it mostly to work except for one thing. I had a look at the tooltip manager and it seems that you can only remove all the tooltips.

Basically in the situation when someone puts text in both the find and replace textboxes, has focus on the replace textbox, hovers their mouse any other button (for example the find button) presses enter (hence triggering the replace function), the find tooltip will disappear.

I'll have a look at how everything works and see if I can get individual tooltips to be deleted.

**Comment by jberk** [comment]
sorry....I did look before opening the ticket.  Thanks for jumping on this

**Comment by thisisnian** [comment]
So what I've done is:

Pulled out each individual tooltip into it's own CompositeDisposible so that each individual tooltip can be  disposed individually.

Can probably be optimised a bit more but for now it should fix this issue. Will do a bit more testing on this.
https://github.com/thisisnian/find-and-replace/commit/8ed7242769d7a8566963ee01619cd8bfb92e1b5b

**Comment by madranet** [comment]
Confirmed also happening on Mac:

![ss 2015-09-18 at 19 08 01](https://cloud.githubusercontent.com/assets/669658/9967146/b81abdc4-5e38-11e5-9a5a-d640a5bc1b9a.png)

A quick work around is to toggle between 'Find in Buffer' and 'Find in Project' by doing a CMD+F and CMD+SHIFT+F dance on the keyboard [or vice versa, as the case may be].

**Comment by dmfilipenko** [comment]
Can't normally use find and replace in editor during this bug

**Comment by eject-it** [comment]
which file do I need to delete to get rid of this stupid thing?
I don't care if it turns off all tooltips. But this is not usable.

**Comment by adambuczynski** [comment]
This does not seem to affect me anymore in the latest version. Have you updated to the latest, @eject-it ?

**Comment by eject-it** [comment]
I am on Atom 1.0.19 and find-and-replace 0.181.0 which I just realised isn't the latest. Don't core packages get updated automatically like all the other stuff?

I always had this bug, but most of the time I can ignore it. This morning it really pissed me off and Google lead me to this issue.

**Comment by 50Wliu** [comment]
> Don't core packages get updated automatically like all the other stuff

No, each Atom version comes bundled with specific versions of core packages.  You can try using Atom 1.1.0-beta1 (https://atom.io/beta) which comes with this fix.
